### PR TITLE
rewards treasury

### PR DESCRIPTION
**Pull request by gitsimon**

### CODE REVIEW

File: pallets/rewards-treasury/src/lib.rs
Comment (RomarQ) [inline pallets/rewards-treasury/src/lib.rs:65]: I understand the rational behind this simplification, but it will cause unexpected behaviors if we update the `Epoch` type with a runtime upgrade.

For example:

The current `Epoch` value is `10000` and I increase it to `15000`. This will cause the burn to happen twice in 15000 blocks (at 10000 and at 15000), while the expected behavior would be for it to happen at 10000 and at 25000.

```suggestion
            if current_block % T::Epoch::get() == 0u16.into() {
```

**Review by RomarQ — CHANGES_REQUESTED** [review]

File: pallets/rewards-treasury/src/lib.rs
Comment (gitsimon) [reply pallets/rewards-treasury/src/lib.rs:65]: True, we could introduce another config to specify a block where only after that block a burn will happen. Only that with this the change of config is still a time-sensitive endeavour and a runtime upgrade might wait for approval. to not further increase complexity I would say we just say it's ok if one epoch is shorter/longer.

Comment (RomarQ) [reply pallets/rewards-treasury/src/lib.rs:65]: An easy way to solve this is to just add a new storage value with the latest block (initial value = 0) where the latest burn happened and then validate `current_block == <Latest_burn> + T::Epoch::get()`

Comment (gitsimon) [reply pallets/rewards-treasury/src/lib.rs]: This is also not robust when `T::Epoch` is shortened; it would never be true again. I did combine the ideas so far

**Pull request merged**
